opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -259,14 +259,15 @@ def end(self, end_time: Optional[int] = None) -> None:
                 raise RuntimeError("Calling end() on a not started span.")
             has_ended = self.end_time is not None
             if not has_ended:
+                if self.status is None:
+                    self.status = Status(canonical_code=StatusCanonicalCode.OK)
```

Comment:
is it possible for a status to be none, even if an end_time exists?

Reply:
I think we should not permit that. I think that `self.end_time` is a fine indicator that the span has ended (we might want to have a more clear attribute, like `self.is_ended`). The idea here is that every ended span should have a status. I'll look into this to make sure that I am not leaving a loophole where that situation (having an ended span with `None` status) can happen.

Reply:
Ok, `end_time` is part of our public API. What we don't want is a user setting this manually to some value while leaving the span status being `None` in an unended weird state. I'll move this into a `property` so that it can only be set by calling `end` (same thing with `start_time`). I'll move this into another PR.

Reply:
I am addressing this issue in #363.